spec/support/pages/organization_dashboard_page.rb

```diff
@@ -0,0 +1,188 @@
+  end
+
+  def has_add_storage_location_call_to_action?
+    has_selector? "#org-stats-call-to-action-storage-locations"
+  end
+
+  def has_diaper_drives_section?
+    has_selector? diaper_drives_selector
+  end
+
+  def has_distributions_section?
+    has_selector? distributions_section_selector
+  end
+
+  def has_getting_started_guide?
+    has_selector? "#getting-started-guide"
+  end
+
+  def has_manufacturers_section?
+    has_selector? manufacturers_section_selector
+  end
+
+  def has_organization_logo?
+    has_selector? org_logo_selector
+  end
+
+  def manufacturers_total_donations
+    within manufacturers_section do
+      parse_formatted_integer find(".total_received_donations").text
+    end
+  end
+
+  def num_manufacturers_donated
+    within manufacturers_section do
+      # the span contains something like "1 Manufacturer" or "28 Manufacturers"
+      # Strip out the number
+      find(".num_manufacturers_donated").text.match(/^\d+/).to_s.to_i
+    end
+  end
+
+  def organization_logo_filepath
+    find(org_logo_selector).native[:src]
+  end
+
+  def recent_diaper_drive_donation_links
+    within diaper_drives_section do
+      all(".donation a").map(&:text)
+    end
+  end
+
+  def recent_distribution_links
+    within distributions_section do
+      all(".distribution a").map(&:text)
+    end
+  end
+
+  def recent_donation_links
+    within donations_section do
+      all(".donation a").map(&:text)
+    end
+  end
+
+  def recent_manufacturer_donation_links
+    within manufacturers_section do
+      all(".manufacturer a").map(&:text)
+    end
+  end
+
+  def recent_purchase_links
+    within purchases_section do
+      all(".purchase a").map(&:text)
+    end
+  end
+
+  def select_date_filter_range(range_name)
+    find("#filters_date_range").click
+    within ".ranges" do
+      find("li[data-range-key='#{range_name}']").click
+    end
+  end
+
+  def summary_section
+    find "#summary"
+  end
+
+  def total_distributed
+    within distributions_section do
+      parse_formatted_integer find(".total_distributed").text
+    end
+  end
+
+  def total_donations
+    within donations_section do
+      parse_formatted_integer find(".total_received_donations").text
+    end
+  end
+
+  def total_inventory
+    within summary_section do
+      parse_formatted_integer find(".total_inventory").text
+    end
+  end
+
+  private
+
+  def diaper_drives_section
+    find diaper_drives_selector
+  end
+
+  def diaper_drives_selector
+    "#diaper_drives"
+  end
+
+  def distributions_section_selector
+    "#distributions"
+  end
+
+  def distributions_section
+    find distributions_section_selector
+  end
+
+  def donations_section
+    find "#donations"
+  end
+
+  def manufacturers_section
+    find manufacturers_section_selector
+  end
+
+  def manufacturers_section_selector
+    "#manufacturers"
+  end
+
+  def org_logo_selector
+    ".organization-logo"
+  end
+
+  def purchases_section
+    find "#purchases"
+  end
+end
```

Comment:
I feel like although this is an improvement, it could be reorganized a bit better. Could we move the methods that are focused around a particular area (e.g. diaper drives or manufacturers) to be grouped together? We could go further and turn them into modules, but I feel like that's overkill.

Reply:
I'm open to suggestions, of course.

For context, this "first cut" is at the "page level"—"just enough," if you will, to support basically/essentially 1:1 parity/conversion of the original `expect()`s' intentions.

In this section, for example, you'll notice that some sections have separate `<name>_section` and `<name>_section_selector` methods while others have the selector directly in the `<name>_section` method.

This was driven by the original examples—some wanted to check the existence of a given selector; others simply assumed its existence. For the former, I extracted the `..._selector` methods for use in the related public `has_<name>_section` method. For the latter, I simply used the selector directly to minimize the steps that someone would have to take reading/reviewing the code to "get it".

Once we agree on & merge this step for converting this spec, I hope/expect further PRs will change(/improve?) the spec's expectations and update/refactor the `*Page` implementations accordingly.

Reply:
Yeah definitely! I'm not suggesting any refactor at this point - literally just talking about reordering the methods in the current class so that particular functionality are in the same area of the code within the class.

Reply:
Yeah +1 on that it feels large but as a first pass I really like it. Love the super clear PORO method names.

Reply:
Would using the "end-less" style help it feel any less lengthy?

I think it can help the one-liner methods feel even more "declarative"—e.g.,

```
def summary_section = find("#summary")
```

Can't say I really like any of these, but for the sake of discussion...
```
def total_distributed =
  within(distributions_section) do
    parse_formatted_integer find(".total_distributed").text
  end
```

```
def total_distributed = within(distributions_section) { parse_formatted_integer find(".total_distributed").text }
```

Reply:
I may be in the minority but I prefer multi line methods. They are easier for my brain to parse.